src/client/containers/chart-detail-page/chart-detailsmartpage-humidity.js

```diff
@@ -0,0 +1,55 @@
+import React, { useContext } from 'react';
+import '../../components/detail-chart/detail-chart.style.css';
+import DetailChart from '../../components/detail-chart/detail-chart.component';
+import SidebarMenu from '../../components/side-navigation/sidebar.component';
+import ProgressBar from '../../components/progress-bar/progress-bar.component';
+import './chart-detail-smartpage.css';
+// import UpdateDateRange from '../../components/update-date-range/update-date-range.component';
+import ChartbarMenu from '../../components/chart-data-buttons/chartbar-buttons/chart-data-button.component';
+import { ChartDataContext } from './chart-detail-context';
+
+const value = [
+  { id: 1, label: 'Last 5 Days' },
+  { id: 2, label: 'Last Week' },
+  { id: 3, label: 'Last Two Weeks' },
+  { id: 4, label: 'Last Month' },
+  { id: 5, label: 'Custom' },
+];
+
+const ChartDetailSmartPageHumidity = () => {
+  const chartvalues = useContext(ChartDataContext);
+  const {
+    boundary,
+    materialName,
+    sensorData,
+    startDate,
+    currentDate,
+    stages,
+    units,
+  } = chartvalues;
+
+  const headingText = materialName.toUpperCase();
+  console.log(sensorData);
+  return (
+    <>
+      <SidebarMenu />
+      <div className="chart-details">
+        <h1>{headingText} GRAPH DETAILS</h1>
+        <ProgressBar
+          startDate={startDate}
+          currentDate={currentDate}
+          stages={stages}
+        />
+        <ChartbarMenu buttons={value} />
```

Comment:
You're missing `defaultSelection` here, e.g. 
```
        <ChartbarMenu buttons={value} defaultSelection={value[0]} />
```

(Although I'm not sure why you're using an object for this prop in `ChartbarMenu`, since it could just be a number. 

P.S. I think you're going to need to store a `dataRange` hook in the `ChartDataContext` as well so you can set it inside `ChartbarMenu` based on which range button is clicked... Then you can trim `sensorData` inside chart-detail-useffect based on that range.

Reply:
we are still working on this  so may be we donot need a default value  selection